core/model/modx/modelement.class.php

```diff
@@ -195,6 +195,36 @@ public function save($cacheFlag = null) {
         return $saved;
     }
 
+    /**
+     * Determine if static files should be automated for current element class.
+     *
+     * @return bool
+     */
+    protected function isStaticFilesAutomated()
+    {
+        $type = '';
+
+        switch ($this->_class) {
+            case 'modTemplate':
+                $type = 'templates';
+                break;
+            case 'modTemplateVar':
+                $type = 'tvs';
+                break;
+            case 'modChunk':
+                $type = 'chunks';
+                break;
+            case 'modSnippet':
+                $type = 'snippets';
+                break;
+            case 'modPlugin':
+                $type = 'plugins';
+                break;
+        }
+
+        return (bool) $this->xpdo->getOption('static_elements_automate_' . $type, null, false);
```

Comment:
What is `type` is an empty string? In this case, all times will be called wrong setting. Make sense fetch setting value only when the type is defined.